src/ReferenceList.php

```diff
@@ -23,7 +27,7 @@
  * @author H. Snater < mediawiki@snater.com >
  * @author Thiemo Mättig
  */
-class ReferenceList extends HashableObjectStorage {
+class ReferenceList extends SplObjectStorage implements Comparable {
```

Comment:
I decided to do it this way round (first remove `HashableObjectStorage`, then refactoring the class). If you prefer it doing the other way round (first removing the SPL stuff) and have some arguments I can also do it that way.